### PR TITLE
Junit report for CI runs

### PR DESCRIPTION
**Pull request by lobziik**
Enable Junit reports for unit-tests run within OCP CI.

### CODE REVIEW

File: hack/go-get-tool.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+     echo "$1 already exists"
+     return
+  fi
+  TMP_DIR=$(mktemp -d)
+  cd "$TMP_DIR"
+  go mod init tmp
+  echo "Downloading $2"
+  GOBIN=$LOCAL_BINARIES_PATH go get "$2"
+  rm -rf "$TMP_DIR"
+  cd -
```
Comment (Fedosin) [inline hack/go-get-tool.sh:22]:
```suggestion
  popd
  rm -rf "$TMP_DIR"
```

File: hack/unit-tests.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ENVTEST_ASSETS_DIR=/tmp/testbin
+ENVTEST_SETUP_SCRIPT=https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/${ENVTEST_VERSION}/hack/setup-envtest.sh
+
+function setupEnvtest() {
+  echo "Envtest version: ${ENVTEST_VERSION}."
+  mkdir -p ${ENVTEST_ASSETS_DIR}
+	test -f ${ENVTEST_ASSETS_DIR}/setup-envtest.sh || curl -sSLo ${ENVTEST_ASSETS_DIR}/setup-envtest.sh ${ENVTEST_SETUP_SCRIPT}
+	source ${ENVTEST_ASSETS_DIR}/setup-envtest.sh
+	fetch_envtest_tools ${ENVTEST_ASSETS_DIR}
+	setup_envtest_env ${ENVTEST_ASSETS_DIR}
```
Comment (Fedosin) [inline hack/unit-tests.sh:20]: Normally you use 2 spaces for indentation, but here we have tabs. could you fix this for consistency?

**Review by Fedosin — APPROVED** [review]
only some small nits

/approve

File: hack/unit-tests.sh (hunk 2 above)
Comment (lobziik) [reply hack/unit-tests.sh]: made it consistent with other sh files there. Didn't noticed that it uses 4 spaces.

File: hack/go-get-tool.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+function go-get-tool() {
+  if [ -f "$1" ] ; then
+     echo "$1 already exists"
+     return
+  fi
+  TMP_DIR=$(mktemp -d)
+  cd "$TMP_DIR"
+  go mod init tmp
+  echo "Downloading $2"
+  GOBIN=$LOCAL_BINARIES_PATH go get "$2"
```
[the remark this answers is not in the document]
Comment (lobziik) [reply hack/go-get-tool.sh]: Hm...
Suggest to keep `go get` there for now and replace this crutches with `go install` in follow up.

**Review by Fedosin — APPROVED** [review]
/lgtm

**Pull request merged**